website/src/main/java/com/google/step/data/CloudSQLManager.java

```diff
@@ -109,4 +109,31 @@ public void createTable(String tableName, List<String> columns) throws SQLExcept
     }
   }
 
+  public ResultSet getOrgsWithNeighbors(String keyword, int offset) throws SQLException{ 
+    String similarTo = (!keyword.isEmpty()) ? 
+        "WHERE (name LIKE '%" + keyword + "%' OR about LIKE '% " + keyword + "%' OR class ='" + keyword + "')" 
+        : "";
+    String preliminaryQuery = String.format("(SELECT * FROM orgTable %sLIMIT " + offset + ", 10)", similarTo);    
+    String query = String.join("\n","SELECT",
+            "preliminary.id, ",
+            "preliminary.link, ",
+            "preliminary.name, " ,
+            "preliminary.about, "  ,
+            "n1.name AS neighbor1, ",
+            "n2.name AS neighbor2, ",
+            "n3.name AS neighbor3, ",
+            "n4.name AS neighbor4",
+        String.format("FROM (%s) AS preliminary", preliminaryQuery),
```

Comment:
sorry it's a bit difficult to understand, could you clarify what "preliminary" is?